dialogflow/stopwatch/Stopwatch/TokenGenerator.swift

```diff
@@ -0,0 +1,90 @@
+//
+// Copyright 2019 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+import Foundation
+import Firebase
+
+class TokenGenerator {
```

Comment:
This doesn't quite generate a token, but rather retrieves a token. So we should come up with another name that more closely resembles what it actually does. 

Use cases:
1. Initial on app load: request new token
- (That token should be used to talk to Dialogflow for all subsequent API calls until the token expires)
2. When the token expires or close to when the token expires (this should be checked before each API call
- This service should be called to retrieve a new OauthToken that the user can keep using from the Firebase Function


Another possible idea in play is that the end user can retrieve tokens from Firestore, before each API call it checks if the token is about to expire.
If it expires in under 10 minutes, request that a new token is generated
If it expires in under 5 minutes, retrieve the latest token from Firestore and see if it expired in ~1hour. Assuming we have already requested that a new token has been requested, if we have not yet requested a new token, do so now then try to retrieve it.

Reply:
As per the suggestion i have renamed the file name from "TokenGenerator.swit" to "TokenReceiver.swift"